query/src/sessions/query_ctx.rs

```diff
@@ -295,37 +294,36 @@ impl QueryContext {
     }
 
     // Get the storage data accessor by config.
-    pub fn get_storage_accessor(&self) -> Result<Arc<dyn DataAccessor>> {
+    // TODO(xuanwo): we can build dal backend only once.
+    pub fn get_storage_accessor(&self) -> Result<Operator> {
         let storage_conf = &self.get_config().storage;
-        let scheme_name = &storage_conf.storage_type;
-        let scheme = StorageScheme::from_str(scheme_name)?;
-        let da: Arc<dyn DataAccessor> = match scheme {
-            StorageScheme::S3 => {
+        let schema_name = &storage_conf.storage_type;
+        let schema = DalSchema::from_str(schema_name)
+            .map_err(|e| ErrorCode::DalTransportError(e.to_string()))?;
+        let da: Arc<dyn Accessor> = match schema {
+            DalSchema::S3 => {
                 let conf = &storage_conf.s3;
-                Arc::new(S3::try_create(
-                    &conf.region,
-                    &conf.endpoint_url,
-                    &conf.bucket,
-                    &conf.access_key_id,
-                    &conf.secret_access_key,
-                    conf.enable_pod_iam_policy,
```

Comment:
@PsiACE  please take a look at this, not sure if we cound temporarily remove `pod_iam_policy`

Reply:
The aws-rust-sdk removed configs like `pod_iam_policy`. According to their README, it seems SDK will load those by default:

![image](https://user-images.githubusercontent.com/5351546/153580253-fa7a2f57-df8f-433c-8671-0d55e7de03c8.png)

https://github.com/awslabs/aws-sdk-rust/

Reply:
I will add `Credential::Env` as default/fallback to load from env instead.

Reply:
We will load credentials from env first and only set credentials if the user has input one. I think the current logic is the same as your said.

Maybe I need to call `aws_config::load_env()` here. But it's an async function, I will figure it out.

Reply:
Resolved in https://github.com/datafuselabs/databend/pull/4081/commits/1bee64ba662bd5e4f04ff911c738a5ed2791933c